src/secex-genprotimgvm-scripts/genprotimg.bu

```diff
@@ -29,6 +29,49 @@ storage:
           genprotimg -V --no-verify -i /var/genprotimg/vmlinuz -r /var/genprotimg/initrd.img  -p /var/genprotimg/parmfile -k /etc/se-hostkeys/ibm-z-hostkey-1 -o /var/genprotimg/se.img
           rm -f /var/genprotimg/signal.file
           bash /var/build/post-script.sh
+    - path: /etc/systemd/system-generators/coreos-genprotimg-generator
+      overwrite: true
+      mode: 0755
+      contents:
+        inline: |
+          #!/bin/bash
+          export PATH="/usr/bin:/usr/sbin:${PATH}"
+          set -euo pipefail
+          . /usr/lib/coreos/generator-lib.sh
+          if [ ! -z $(karg ignition.firstboot) ]; then
+              exit 0
+          fi
+          mkdir -p "${UNIT_DIR}/default.target.wants"
+
+          cat > "${UNIT_DIR}"/var-build.mount << 'EOF'
+          # generated by coreos-genprotimg-generator
+          # Mounts partition with parmfile, pre- and post- scripts used by genprotimg.service
+          [Unit]
+          Description=Genprotimg partition mount
+          Requires=dev-disk-by\x2did-virtio\x2dgenprotimg.device
+          After=dev-disk-by\x2did-virtio\x2dgenprotimg.device
+          [Mount]
+          What=/dev/disk/by-id/virtio-genprotimg
+          Where=/var/build
+          Type=ext4
+          Options=rw,noatime
+          EOF
+          ln -sf "../var-build.mount" "${UNIT_DIR}/default.target.wants/"
+
+          cat > "${UNIT_DIR}/genprotimg.service" << 'EOF'
+          # generated by coreos-genprotimg-generator
+          # Creates secure image by calling genprotimg tool
+          [Unit]
+          Description=Genprotimg executor
+          After=var-build.mount
+          After=multi-user.target
+          [Service]
+          Type=oneshot
+          StandardOutput=journal+console
+          ExecStart=/etc/do_genprotimg
+          ExecStopPost=/usr/bin/systemctl --no-block poweroff
```

Comment:
What is `ExecStopPost` giving us right here? Wouldn't that only execute when the system is shutting down?

Reply:
I guess it's because it's a `oneshot` and doesn't have `RemainAfterExit=yes` then `ExecStopPost` will execute immediately.

I'd be much less confused if this was `ExecStartPost` instead of `ExecStopPost`. It would have the same effect, but be less confusing.